### PR TITLE
[MI-2120] Add handling of card response from VA

### PR DESCRIPTION
**Pull request by ayusht2810**

### CODE REVIEW

File: server/plugin/api.go [hunk 1]
```diff
@@ -45,7 +46,9 @@ func (p *Plugin) initializeAPI() *mux.Router {
 
 func (p *Plugin) checkAuthBySecret(handleFunc http.HandlerFunc) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		if status, err := verifyHTTPSecret(p.getConfiguration().WebhookSecret, r.FormValue("secret")); err != nil {
+		// Replace all occurrences of " " with "+".
+		secret := strings.ReplaceAll(r.FormValue("secret"), " ", "+")
```
Comment (Kshitij-Katiyar) [inline server/plugin/api.go:50]: make a constant for secret

File: server/plugin/virtual_agent.go [hunk 2]
```diff
@@ -76,15 +77,46 @@ type TopicPickerControl struct {
...
+	UIType       string `json:"uiType"`
+	Group        string `json:"group"`
+	Data         string `json:"data"`
+	TemplateName string `json:"templateName"`
+}
+
+type OutputCardRecordData struct {
+	SysID            string         `json:"sys_id"`
+	Subtitle         string         `json:"subtitle"`
+	DataNowSmartLink string         `json:"dataNowSmartLink"`
```
Comment (Kshitij-Katiyar) [inline server/plugin/virtual_agent.go:89]: just a question the casing for json variables is different there is snake case for dataNowSmartLink and not for other variables.

File: server/plugin/virtual_agent.go [hunk 3]
```diff
@@ -219,15 +251,45 @@ func (p *Plugin) ProcessResponse(data []byte) error {
...
+					return err
+				}
+
+				if _, err = p.DMWithAttachments(userID, p.CreateOutputCardVideoAttachment(&data)); err != nil {
+					return err
+				}
+
+				channel, err := p.API.GetDirectChannel(userID, p.botUserID)
+				if err != nil {
+					p.API.LogInfo("Couldn't get bot's DM channel with user", "userID", user)
```
Comment (Kshitij-Katiyar) [inline server/plugin/virtual_agent.go:277]: for this log i think we should not print the whole user just the userID or username is enough.

File: server/plugin/virtual_agent.go [hunk 4]
```diff
@@ -219,15 +251,45 @@ func (p *Plugin) ProcessResponse(data []byte) error {
...
+					return err
+				}
+
+				URL := url.URL{RawQuery: data.Link}
+				targetURL := URL.Query().Get(VideoQueryParam)
```
Comment (Kshitij-Katiyar) [inline server/plugin/virtual_agent.go:282]: we can direct use this variable

File: server/plugin/virtual_agent.go (hunk 2 above)
Comment (ayusht2810) [reply server/plugin/virtual_agent.go:89]: yes, we are getting different casing in the response

File: server/plugin/virtual_agent.go [hunk 5]
```diff
@@ -219,15 +251,44 @@ func (p *Plugin) ProcessResponse(data []byte) error {
...
+				}
+
+				channel, err := p.API.GetDirectChannel(userID, p.botUserID)
+				if err != nil {
+					p.API.LogInfo("Couldn't get bot's DM channel with user", "userID", userID)
+					return err
+				}
+
+				URL := url.URL{RawQuery: data.Link}
+				p.Ephemeral(userID, channel.Id, URL.Query().Get(VideoQueryParam))
```
Comment (raghavaggarwal2308) [inline server/plugin/virtual_agent.go:282]: What is this code doing?

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: server/plugin/api.go [hunk 6]
```diff
@@ -45,7 +46,9 @@ func (p *Plugin) initializeAPI() *mux.Router {
 
 func (p *Plugin) checkAuthBySecret(handleFunc http.HandlerFunc) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		if status, err := verifyHTTPSecret(p.getConfiguration().WebhookSecret, r.FormValue("secret")); err != nil {
+		// Replace all occurrences of " " with "+".
```
Comment (Kshitij-Katiyar) [inline server/plugin/api.go:49]:
```suggestion
		// Replace all occurrences of " " with "+" in WebhookSecret.
```

**Review by Kshitij-Katiyar — APPROVED** [review]

File: server/plugin/virtual_agent.go (hunk 5 above)
Comment (ayusht2810) [reply server/plugin/virtual_agent.go:282]: we post the video in a preview using ephemeral post

Comment (raghavaggarwal2308) [reply server/plugin/virtual_agent.go:282]: Why ephemeral?

Comment (ayusht2810) [reply server/plugin/virtual_agent.go]: updated

File: server/plugin/virtual_agent.go [hunk 7]
```diff
@@ -272,14 +272,10 @@ func (p *Plugin) ProcessResponse(data []byte) error {
...
-				if err != nil {
-					p.API.LogInfo("Couldn't get bot's DM channel with user", "userID", userID)
-					return err
-				}
-
 				URL := url.URL{RawQuery: data.Link}
-				p.Ephemeral(userID, channel.Id, URL.Query().Get(VideoQueryParam))
+				p.dm(userID, &model.Post{
+					Message: URL.Query().Get(VideoQueryParam),
+				})
```
Comment (raghavaggarwal2308) [inline server/plugin/virtual_agent.go]: Check for error

**Review by raghavaggarwal2308 — APPROVED** [review]

File: server/plugin/virtual_agent.go (hunk 7 above)
Comment (ayusht2810) [reply server/plugin/virtual_agent.go]: checking

**Pull request closed without merging**